common/gratia/common/condor_ce.py

```diff
@@ -142,19 +142,22 @@ def createCertinfoXML(classad):
     uniqID.appendChild(dom.createTextNode(unique_id))
 
     # Sample: x509UserProxyVOName = "cms"
-    voStr = classad.get("x509UserProxyVOName", "")
+    voStr = classad.get("x509UserProxyVOName", "") or \
+            classad.get("AuthTokenIssuer", "")
     vo = dom.createElement("VO")
     vo.appendChild(dom.createTextNode(voStr))
 
     # Sample: x509UserProxyFirstFQAN = "/cms/Role=NULL/Capability=NULL"
-    fqanStr = classad.get("x509UserProxyFirstFQAN", "")
+    fqanStr = classad.get("x509UserProxyFirstFQAN", "") or \
+              classad.get("AuthTokenIssuer", "")
     fqan = dom.createElement("FQAN")
     fqan.appendChild(dom.createTextNode(fqanStr))
 
     # Sample: x509userproxysubject = "/C=TW/O=AP/OU=GRID/CN=Nitish Dhingra 142746"
-    x509Str = classad.get("x509userproxysubject", "")
-    x509 = dom.createElement("DN")
-    x509.appendChild(dom.createTextNode(x509Str))
+    dnStr = classad.get("x509userproxysubject", "") or \
+            classad.get("AuthTokenSubject", "")
+    dn = dom.createElement("DN")
+    dn.appendChild(dom.createTextNode(dnStr))
```

Comment:
Won't all the `classad.get(...) or` expressions always result in the `x509*` attr value or the empty string?

Reply:
Yes.  It defaults to the x509 value, if present, otherwise the empty string evaluates as False in a boolean context, so the short-circuit `or` will fall back to the AuthToken\* value

Reply:
Oh right